tests/unit/browser/test_caret.py

```diff
@@ -61,16 +63,24 @@ def check(self, expected, *, strip=False):
                     selection = selection.strip()
                 assert selection == expected
                 return
+            elif not selection and not expected:
+                return
 
             self._qtbot.wait(50)
 
+        assert False, 'Failed to get selection!'
+
```

Comment:
I wanted to ask about that, I also noticed failing tests on Travis. But unfortunatelly I do not have access to any Mac computer to investigate it. One test on MacOS is also failing because of line-selection (bug?). I will try to find out why, but as I already mentioned, can not guarantee to find a solution.